contrib/codegen/funcs/funcs.go

```diff
@@ -80,37 +80,43 @@ func clientImportPath(grp importedGroup) string {
 }
 
 // pass empty string if clients live in the same go module as the type definitions
-func selectResources(clientModule string, groups []model.Group, resourcesToSelect map[schema.GroupVersion][]string) []importedGroup {
+func selectResources(groups map[string][]model.Group, resourcesToSelect map[schema.GroupVersion][]string) []importedGroup {
 	var selectedResources []importedGroup
-	for _, group := range groups {
-		resources := resourcesToSelect[group.GroupVersion]
-		if len(resources) == 0 {
-			continue
-		}
-		filteredGroup := group
-		filteredGroup.Resources = nil
-
-		isResourceSelected := func(kind string) bool {
-			for _, resource := range resources {
-				if resource == kind {
-					return true
+	for clientModule, groups := range groups {
+		for _, group := range groups {
+			resources := resourcesToSelect[group.GroupVersion]
+			if len(resources) == 0 {
+				continue
+			}
+			filteredGroup := group
+			filteredGroup.Resources = nil
+
+			isResourceSelected := func(kind string) bool {
+				for _, resource := range resources {
+					if resource == kind {
+						return true
+					}
 				}
+				return false
 			}
-			return false
-		}
 
-		for _, resource := range group.Resources {
-			if !isResourceSelected(resource.Kind) {
-				continue
+			for _, resource := range group.Resources {
+				if !isResourceSelected(resource.Kind) {
+					continue
+				}
+				filteredGroup.Resources = append(filteredGroup.Resources, resource)
 			}
-			filteredGroup.Resources = append(filteredGroup.Resources, resource)
-		}
 
-		selectedResources = append(selectedResources, importedGroup{
-			Group:    filteredGroup,
-			GoModule: clientModule,
-		})
+			selectedResources = append(selectedResources, importedGroup{
+				Group:    filteredGroup,
+				GoModule: clientModule,
+			})
+		}
 	}
 
+	sort.SliceStable(selectedResources, func(i, j int) bool {
+		return selectedResources[i].GoModule < selectedResources[j].GoModule
+	})
```

Comment:
Should we use the `GVK.String()` here instead? Go module isn't guaranteed to be different.

Reply:
resources are randomized across their gomodule, so sorting by gomodule should be sufficient